### PR TITLE
Feature/search range

### LINKED ISSUE #81: FS: Search Contests
- Create a search bar on FE with start/end date pickers
- Modify BE contest controllers
- Integrate search functionality

### PR DESCRIPTION
**Pull request by brian-glovebox**
### What this PR does (required):
- Able to filter contests table by deadline date chosen

### Screenshots / Videos (required):
Screen Shot 2021-07-27 at 9 41 06 PM

### Any information needed to test this feature (required):
- Choose a date from date picker and the table will rerender.

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/controllers/contest.js [hunk 1]
```diff
@@ -65,4 +69,42 @@ exports.getAllContestsByUser = asyncHandler(async (req, res) => {
   } catch (err) {
     res.status(500).json(err);
   }
+})
+
+exports.getContestsByDeadlineDate = asyncHandler(async (req, res) => {
```
Comment (rajivtitus) [inline server/controllers/contest.js]: Maybe use the `getAllContests` controller to implement this logic? By doing so, we will only have one route and one controller 👍🏽

File: client/src/pages/Discovery/Discovery.tsx [hunk 2]
```diff
@@ -38,13 +43,27 @@ export default function Discovery(): JSX.Element {
...
         setContests(allContests.contests);
       } else {
-        console.log('No Contests Found');
         new Error('Could Not Get Contests');
       }
     }
     getAll();
   }, []);
 
+  useEffect(() => {
```
Comment (rajivtitus) [inline client/src/pages/Discovery/Discovery.tsx:53]: Please see comment below

File: server/controllers/contest.js (hunk 1 above)
Comment (brian-glovebox) [reply server/controllers/contest.js]: a little unclear what you mean here, one controller requires a query parameter and one does not.

Comment (rajivtitus) [reply server/controllers/contest.js]: You can add the query parameters to the `getAllContests` controller. If the parameters have not been specified (i.e undefined), you can fetch all the contests. On the other hand, if paraments have been specified, you can fetch contests based on the date range.

Note: On the FE, we may need to have a button to reset the date filters.

Comment (brian-glovebox) [reply server/controllers/contest.js]: sounds good, will fix this after work today

File: client/src/pages/Discovery/Discovery.tsx (hunk 2 above)
Comment (brian-glovebox) [reply client/src/pages/Discovery/Discovery.tsx:53]: added changes requested along with the button to reset the filter.

**Pull request merged**